src/uwtools/cli.py

```diff
@@ -425,6 +429,114 @@ def _dispatch_fv3(args: Args) -> bool:
     )
 
 
+# Mode mpas
+
+
+def _add_subparser_mpas(subparsers: Subparsers) -> ModeChecks:
+    """
+    Subparser for mode: mpas
+
+    :param subparsers: Parent parser's subparsers, to add this subparser to.
+    """
+    parser = _add_subparser(subparsers, STR.mpas, "Execute MPAS tasks")
+    _basic_setup(parser)
+    subparsers = _add_subparsers(parser, STR.action, STR.task.upper())
+    return {
+        task: _add_subparser_mpas_task(subparsers, task, helpmsg)
+        for task, helpmsg in uwtools.api.mpas.tasks().items()
+    }
+
+
+def _add_subparser_mpas_task(subparsers: Subparsers, task: str, helpmsg: str) -> ActionChecks:
+    """
+    Subparser for mode: mpas <task>
+
+    :param subparsers: Parent parser's subparsers, to add this subparser to.
+    :param task: The task to add a subparser for.
+    :param helpmsg: Help message for task.
+    """
+    parser = _add_subparser(subparsers, task, helpmsg.rstrip("."))
+    required = parser.add_argument_group(TITLE_REQ_ARG)
+    _add_arg_config_file(group=required, required=True)
+    _add_arg_cycle(required)
+    optional = _basic_setup(parser)
+    _add_arg_batch(optional)
+    _add_arg_dry_run(optional)
+    _add_arg_graph_file(optional)
+    checks = _add_args_verbosity(optional)
+    return checks
+
+
+def _dispatch_mpas(args: Args) -> bool:
+    """
+    Dispatch logic for mpas mode.
+
+    :param args: Parsed command-line args.
+    """
+    return uwtools.api.mpas.execute(
+        task=args[STR.action],
+        config=args[STR.cfgfile],
+        cycle=args[STR.cycle],
+        batch=args[STR.batch],
+        dry_run=args[STR.dryrun],
+        graph_file=args[STR.graphfile],
+    )
+
+
+# Mode mpas_init
+
+
+def _add_subparser_mpas_init(subparsers: Subparsers) -> ModeChecks:
+    """
+    Subparser for mode: mpas_init
+
+    :param subparsers: Parent parser's subparsers, to add this subparser to.
+    """
+    parser = _add_subparser(subparsers, STR.mpasinit, "Execute MPASInit tasks")
+    _basic_setup(parser)
+    subparsers = _add_subparsers(parser, STR.action, STR.task.upper())
+    return {
+        task: _add_subparser_mpas_init_task(subparsers, task, helpmsg)
+        for task, helpmsg in uwtools.api.mpas_init.tasks().items()
+    }
+
+
+def _add_subparser_mpas_init_task(subparsers: Subparsers, task: str, helpmsg: str) -> ActionChecks:
+    """
+    Subparser for mode: mpas_init <task>
+
+    :param subparsers: Parent parser's subparsers, to add this subparser to.
+    :param task: The task to add a subparser for.
+    :param helpmsg: Help message for task.
+    """
+    parser = _add_subparser(subparsers, task, helpmsg.rstrip("."))
+    required = parser.add_argument_group(TITLE_REQ_ARG)
+    _add_arg_config_file(group=required, required=True)
+    _add_arg_cycle(required)
+    optional = _basic_setup(parser)
```

Comment:
@maddenp-noaa I've updated `config` to be `Optional` within the two `mpas_init.py` files, but I'm curious why it doesn't cause an issue here. I will update it, I just found it to be oddly inconsistent.

Reply:
I think `--config-file` should be optional here, too, as for all (I think) the other drivers: The config can be read from `stdin` as well as from a file.